netbox/netbox/settings.py

```diff
@@ -288,6 +293,30 @@ def _setting(name, default=None):
     'utilities.auth_backends.ViewExemptModelBackend',
 ]
 
+# GitHub repository for version check
+if GITHUB_REPOSITORY_API:
+    GITHUB_REPOSITORY_API = GITHUB_REPOSITORY_API.rstrip('/')
+    try:
+        scheme, netloc, path, query, fragment = urlsplit(GITHUB_REPOSITORY_API)
+    except ValueError:
+        raise ImproperlyConfigured("GITHUB_REPOSITORY_API must be a valid URL")
+
+    if scheme not in ('http', 'https'):
+        raise ImproperlyConfigured("GITHUB_REPOSITORY_API must be a valid http:// or https:// URL")
+
+    if not re.fullmatch(r'/repos/[A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+', path):
+        raise ImproperlyConfigured(
+            "GITHUB_REPOSITORY must contain the base URL of the GitHub API in a form like "
+            "'https://api.github.com/repos/<owner>/<repository>'"
+        )
+
+    if query or fragment:
+        raise ImproperlyConfigured("GITHUB_REPOSITORY_API may not contain a query or fragment")
```

Comment:
This error message is unlikely to make sense to the user out of context. I don't even know what would trigger it and I'm looking at the code right now. What are we checking for?

Reply:
This checks that the URL provided doesn't contain a query (`URL?q=bla`) or a fragment (`URL#somewhere`).